scripts/o3de/tests/test_repo_properties.py

```diff
@@ -310,7 +395,11 @@ def get_json_data(object_typename: str,
                 assert self.repo_json.data.get('origin') == 'o3de'
 
                 for gem in self.repo_json.data.get('gems_data', []):
-                    assert gem in expected_gems
+                    if release_archive_path:
+                        zip_path = release_archive_path / pathlib.Path(gem['download_source_uri']).name
+                        download.validate_downloaded_zip_sha256(gem, zip_path, "gem.json")
```

Comment:
do we need to assert download.validate_downloaded_zip_sha256 returns true or something?

Reply:
The function will raise an error if it fails, since its not mocked, the original function won't pass

Reply:
Spoke offline with @VeryKaylin-player1.
Unit test will error inside that method and so will fail (as expected)